stdlib/lazy.mli

```diff
@@ -139,3 +139,91 @@ val force_val : 'a t -> 'a
 
     @raise Undefined (see {!Undefined}).
 *)
+
+module Concurrent_blocking : sig
+  (** Simple mutex-protected concurrent lazy thunks.
+
+      This implementation has two downsides:
+      - It is less optimized than [Lazy.t].
+      - It uses a standard library {!Mutex} to wait on concurrent
+        initialization, so it does not interoperate well with user-level
+        fiber/task abstractions.
+
+      A typical use-case is optional library initialization code that
+      is moderately expensive, or acquires resources. The library
+      author does not want to do this work on startup, because it may
+      not be needed, but using ['a Lazy.t] is incorrect if the library
+      may be used in concurrent settings. ['a Lazy.Concurrent_blocking.t]
+      can be used, as long as the blocking behavior is acceptable.
+
+     {b Note}: ['a Lazy.t] contains a protection against recursively
+     forcing a thunk, it will raise {!Undefined}. On the other hand,
+     ['a Lazy.Concurrent_blocking.t] will try to lock the mutex
+     recursively, which will raise [Sys_error].
+
+     See {{!examples} the examples} below.
+
+     @since 5.5
+  *)
+
+  type 'a t
+  (* A value of type ['a Lazy.Atomic_repeating.t] is similar to a value
+     of type ['a Lazy.t], it represents a deferred computation, but it can
+     safely be used in concurrent settings.
+
+     If a calling domain attempts to {!force} a value that is already
+     being forced, the calling domain is not suspended. Instead, the
+     computation of the value will be repeated on the calling
+     domain. In other words, [Atomic_repeating.t] can duplicate
+     computations.
+
+     The implementation ensures that all call to {!force} return the
+     same value or raise the same exception: if a repeated terminates
+     on a result, its value or exception will be discarded.
+  *)
```

Comment:
It looks like this documentation comment has not been updated from the `Atomic_repeating` version.

Reply:
Woops! Fixed.